storage/mongo/storage.go

```diff
@@ -0,0 +1,251 @@
+package mongo
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/netlify/netlify-auth/conf"
+	"github.com/netlify/netlify-auth/crypto"
+	"github.com/netlify/netlify-auth/models"
+	"github.com/pkg/errors"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+type logger struct {
+	entry *logrus.Entry
+}
+
+func (l logger) Output(depth int, s string) error {
+	l.entry.Print(s)
+	return nil
+}
+
+type Connection struct {
+	session *mgo.Session
+	db      *mgo.Database
+	config  *conf.Configuration
+}
+
+func (conn *Connection) Close() error {
+	conn.session.Close()
+	return nil
+}
+
+func (conn *Connection) Automigrate(models ...interface{}) error {
+	return nil
+}
+
+func (conn *Connection) CreateUser(user *models.User) error {
+	c := conn.db.C(user.TableName())
+	if err := c.Insert(user); err != nil {
+		return errors.Wrap(err, "Error creating user")
+	}
+
+	if !conn.config.JWT.AdminGroupDisabled {
+		v, err := c.Find(bson.M{"_id": bson.M{"$ne": user.ID}}).Count()
+		if err != nil {
+			return errors.Wrap(err, "Error making user an admin")
+		}
+
+		if v == 0 {
+			user.SetRole(conn.config.JWT.AdminGroupName)
+			if err := c.Update(bson.M{"_id": user.ID}, bson.M{"$set": user}); err != nil {
+				return errors.Wrap(err, "Error making user an admin")
+			}
+		}
+	}
+
+	return nil
+}
+
+func (conn *Connection) FindUserByEmail(email string) (*models.User, error) {
+	return conn.findUser(bson.M{"email": email})
+}
+
+func (conn *Connection) FindUserByID(id string) (*models.User, error) {
+	return conn.findUser(bson.M{"_id": id})
+}
+
+func (conn *Connection) FindUserByVerificationToken(verificationType models.VerifyType, token string) (*models.User, error) {
+	return conn.findUser(bson.M{string(verificationType): token})
+}
+
+func (conn *Connection) FindUserWithRefreshToken(token string) (*models.User, *models.RefreshToken, error) {
+	refreshToken := &models.RefreshToken{}
+	rc := conn.db.C(refreshToken.TableName())
+
+	if err := rc.Find(bson.M{"token": token}).One(refreshToken); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, nil, models.RefreshTokenNotFoundError{}
```

Comment:
instead of having a not found per model, it seems simpler to have a general not found error in the models package.